src/h3lib/lib/h3Index.c

```diff
@@ -211,37 +217,36 @@ H3Index makeDirectChild(H3Index h, int cellNumber) {
 /**
  * h3ToChildren takes the given hexagon id and generates all of the children
  * at the specified resolution storing them into the provided memory pointer.
- * It's assumed that maxH3ToChildrenSize was used to determine the allocation.
+ * It's assumed that cellToChildrenSize was used to determine the allocation.
  *
  * @param h H3Index to find the children of
  * @param childRes int the child level to produce
  * @param children H3Index* the memory to store the resulting addresses in
  */
 void H3_EXPORT(h3ToChildren)(H3Index h, int childRes, H3Index* children) {
-    int parentRes = H3_GET_RESOLUTION(h);
-    if (!_isValidChildRes(parentRes, childRes)) {
-        return;
-    } else if (parentRes == childRes) {
-        *children = h;
-        return;
-    }
-    int bufferSize = H3_EXPORT(maxH3ToChildrenSize)(h, childRes);
-    int bufferChildStep = (bufferSize / 7);
-    int isAPentagon = H3_EXPORT(h3IsPentagon)(h);
-    for (int i = 0; i < 7; i++) {
-        if (isAPentagon && i == K_AXES_DIGIT) {
-            H3Index* nextChild = children + bufferChildStep;
-            while (children < nextChild) {
-                *children = H3_NULL;
-                children++;
-            }
-        } else {
-            H3_EXPORT(h3ToChildren)(makeDirectChild(h, i), childRes, children);
-            children += bufferChildStep;
-        }
+    Iter_Child iter = iterInitParent(h, childRes);
+
+    for (int64_t i = 0; iter.h; i++, iterStepChild(&iter)) {
+        children[i] = iter.h;
     }
 }
 
+/* Zero out index digits from start to end, inclusive.
+ * No-op if start > end.
+ **/
+H3Index _zero_index_digits(H3Index h, int start, int end) {
+    if (start > end) return h;
+
+    uint64_t m = ~0;
+
+    m <<= 3 * (end - start + 1);
+    m = ~m;
+    m <<= 3 * (15 - end);
+    m = ~m;
```

Comment:
I'd be inclined to use the for loop approach here for simplicity, but I get that using the bitwise math is theoretically a little faster.